prowler/providers/aws/services/elbv2/elbv2_service.py

```diff
@@ -50,6 +53,111 @@ def __describe_load_balancers__(self, regional_client):
             logger.error(
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
+    
+    def __describe_target_groups__(self, regional_client):
+        logger.info("ELBv2 - Describing target groups...")
+        try:
+            for lb in self.loadbalancersv2:
+                try:
+                    if (
+                        lb.scheme == "internet-facing"
+                        and lb.type == "application"
+                        and len(lb.security_groups) > 0
+                    ):
+                        describe_target_groups_paginator = (
+                            regional_client.get_paginator("describe_target_groups")
+                        )
+                        for page in describe_target_groups_paginator.paginate(
+                            LoadBalancerArn=lb.arn
+                        ):
+                            for target_group in page["TargetGroups"]:
+                                for target_health in regional_client.describe_target_health(
+                                    TargetGroupArn=target_group["TargetGroupArn"]
+                                )[
+                                    "TargetHealthDescriptions"
+                                ]:
+                                    tg = TargetGroups(
+                                        name=target_group["TargetGroupName"],
+                                        arn=target_group["TargetGroupArn"],
+                                        target_type=target_group["TargetType"],
+                                        target=target_health["Target"]["Id"],
+                                        public=(
+                                            True
+                                            if lb.scheme == "internet-facing"
+                                            and lb.type == "application"
+                                            and len(lb.security_groups) > 0
+                                            else False
```

Comment:
This is never going to be false due to first if of the for

Reply:
fixed